internal/terraform/context_plan.go

```diff
@@ -334,11 +334,11 @@ func (c *Context) destroyPlan(config *configs.Config, prevRunState *states.State
 	// must coordinate with this by taking that action only when c.skipRefresh
 	// _is_ set. This coupling between the two is unfortunate but necessary
 	// to work within our current structure.
-	if !opts.SkipRefresh {
+	if !opts.SkipRefresh && !prevRunState.Empty() {
 		log.Printf("[TRACE] Context.destroyPlan: calling Context.plan to get the effect of refreshing the prior state")
 		normalOpts := *opts
-		normalOpts.Mode = plans.NormalMode
-		refreshPlan, refreshDiags := c.plan(config, prevRunState, &normalOpts)
+		normalOpts.Mode = plans.RefreshOnlyMode
```

Comment:
silly nit: this variable name `normalOpts` was originally meaning "options with normal mode" (as opposed to the options with destroy mode that we're deriving this from).

Should we rename this `refreshOpts` instead now, to reflect this new meaning? (In retrospect that name probably would've made more sense even before this change, since it is the opts we use for refreshing.)

Reply:
Of course 🤦 I'll update